plugins/footnote/spec/system/rich_editor_extension_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+describe "Composer - ProseMirror editor - Footnote extension", type: :system do
+  fab!(:user) { Fabricate(:user, refresh_auto_groups: true) }
+  let(:cdp) { PageObjects::CDP.new }
+  let(:composer) { PageObjects::Components::Composer.new }
+  let(:rich) { composer.rich_editor }
+
+  before do
+    sign_in(user)
+    SiteSetting.rich_editor = true
+  end
+
+  def open_composer_and_toggle_rich_editor
+    page.visit "/new-topic"
+    expect(composer).to be_opened
+    composer.toggle_rich_editor
+  end
+
+  describe "pasting content" do
+    it "converts inline footnotes" do
+      cdp.allow_clipboard
+      open_composer_and_toggle_rich_editor
+      rich.click
+
+      cdp.write_clipboard <<~MARKDOWN
+        What is this? ^[multiple inline] ^[footnotes]
+      MARKDOWN
+      page.send_keys([SystemHelpers::PLATFORM_KEY_MODIFIER, "v"])
```

Comment:
FWIW I made a copy_paste helper for this kind of write to clipboard + ctrl-v action:

https://github.com/discourse/discourse/blob/fde781d0ddce3982dd185a46f8be3805364d62e8/spec/system/page_objects/cdp.rb#L59-L63